py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,811 @@
+    icons: A dictionary relating weather code to icon
+        See https://openweathermap.org/weather-conditions for a complete list
+        of supported icons. This will fall-back to the listed icon if there is
+        no specific icon present. However, options included here take precedent
+        over the above 'icon_{...}' options.
+        There are multiple ways to specify individual icons based on the id:
+            * Use the key '601' to reference the condition with id = 601
+              (snow)
+            * Use the key '230_232' to reference a span of conditions
+              inclusive, in this case conditions (230, 231, 232) (thunderstorm
+              with drizzle)
+        (default None)
+    lang: An ISO 639-1 code for your language (two letters)
+        (default 'en')
+    location: A tuple of floats describing the desired weather location
+        The tuple should follow the form (latitude, longitude), and if set,
+        implicitly disables the Timezone API for determining location.
+        (default None)
+    offset_gmt: A string describing the offset from GMT (UTC)
+        The string should follow the format '+12:34', where the first
+        character is either '+' or '-', followed by the offset in hours,
+        then the offset in minutes. If this is set, it disables the
+        automatic timezone detection from the Timezone API.
+        (default None)
+    request_timeout: The timeout in seconds for contacting the API's.
+        (default 10)
+    thresholds: Configure temperature colors based on limits
+        The numbers specified inherit the unit of the temperature as configured.
+        The default below is intended for Fahrenheit. If the set value is empty
+        or None, the feature is disabled. You can specify this parameter using a
+        dictionary:
+            * Keys are names. You have the option of 'current', 'min', 'max',
+              or 'all' to specify a threshold. The first three are tied to the
+              various temperature values, the last sets the same threshold for
+              all outputs. If both 'all' and one of the first three are set
+              (lets say 'min' for this example), the threshold will default to
+              be the value in 'min', not 'all'. This goes for any configuration
+            * The values are lists of pairs, with temperature (in the
+              configured unit) as the first and the color as the second
+            * To use the thresholds color, place '\?color=all' in the
+              formatting string for temperature, replacing 'all' with any of
+              the valid threshold names for different coloring effects
+            * To have smooth transitions between colors, consider setting the
+              'gradients' configuration parameter to 'True', either in the
+              global configuration, or in the module configuration!
+        (default {'all': [(-100, '#0FF'), (0, '#00F'), (50, '#0F0'), (150, '#FF0')]})
```

Comment:
This trips on a list.
`thresholds = [(-100, '#0FF'), (0, '#00F'), (50, '#0F0'), (150, '#FF0')]`
```
AttributeError: 'list' object has no attribute 'keys'
  File "/home/chris/src/py3status/py3status/module.py", line 132, in prepare_module
    self.module_class.post_config_hook()
  File "/home/chris/src/py3status/py3status/modules/weather_owm.py", line 417, in post_config_hook
    if set(self.thresholds.keys()) > THRESHOLDS_NAMES:
2017-09-10 17:04:40 INFO terminating module weather_owm
```
@tobes Maybe you want to review this part because you're more familiar with thresholds? This is making things either easier or complicated when it's going to play favorites with the temperatures.

Reply:
I'll take a look, but a quick skim makes me think this is not a problem we probably just need to remove this bit of code anyway.

Reply:
Friendly ping here, what should I do to help out? :)

Reply:
Sorry things have been a bit busy the last few weeks